cunumeric/array.py

```diff
@@ -3663,7 +3663,7 @@ def _perform_binary_reduction(
         two,
         dtype,
         extra_args=None,
-    ):
+    ) -> Any:
```

Comment:
Some predicates return a bool
```
cunumeric/module.py: note: In function "allclose":
cunumeric/module.py:3785:12: error: Incompatible return value type (got "ndarray", expected "bool")  [return-value]
        return ndarray._perform_binary_reduction(
               ^
cunumeric/module.py: note: In function "array_equal":
cunumeric/module.py:3883:12: error: Incompatible return value type (got "ndarray", expected "bool")  [return-value]
        return ndarray._perform_binary_reduction(
```
Does `_perform_binary_reduction` return a bool ndarray  in this case? If so should `allclose`, etc do an adaptation to bool? Or should the docs/types for `allclose` be changed instead?

Reply:
Well, [`numpy.allclose`] does indeed return `bool` but we do not:

```
>>> shape = (100,)
>>> filter_shape = (5,)
>>> a = num.random.randint(0, 5, shape)
>>> v = num.random.randint(0, 5, filter_shape)
>>> anp = a.__array__()
>>> vnp = v.__array__()
>>> out = num.convolve(a, v, mode="same")
>>> out_np = np.convolve(anp, vnp, mode="same")
>>> num.allclose(out, out_np)
array(True)
```

@magnatelee what do you want to do here? I propose  change reduction return to `ndarray` and cast the result to `bool` in `allclose`, etc before returning.

Reply:
I think we want to revert the change and update the docstrings instead. we've been using scalar arrays as deferred scalars to avoid any blocking behavior and those casting calls would introduce blocking.

Reply:
https://github.com/nv-legate/cunumeric/pull/413/commits/2653a152761cf2809e1c91b785fef0fab2dc8943